ubr/utils.py

```diff
@@ -146,44 +160,40 @@ def isint(x):
 def choose(prompt, choices, label_fn=None):
     try:
         if label_fn:
-            labels = zip(choices, map(label_fn, choices)) # ll: [(/foo/bar/baz, baz), (/foo/bar/bup, bup)]
+            labels = list(zip(choices, list(map(label_fn, choices)))) # ll: [(/foo/bar/baz, baz), (/foo/bar/bup, bup)]
         else:
-            labels = zip(choices, choices)
+            labels = list(zip(choices, choices))
         idx = enumerated(choices) # ll: {1: /foo/bar/baz, 2: /foo/bar/bup}
 
         while True:
             # present menu
             for i, pair in enumerate(labels):
-                print '%s: %s' % (i + 1, pair[1])
-            print
+                print('%s: %s' % (i + 1, pair[1]))
+            print()
 
             # prompt user
-            uin = raw_input(prompt)
+            uin = input(prompt)
 
             # hygeine
             if not uin or not uin.strip():
-                print 'a choice is required (ctrl-c to quit)'
+                print('a choice is required (ctrl-c to quit)')
                 continue
             if not isint(uin):
-                print 'a -numeric- choice is required (ctrl-c to quit)'
+                print('a -numeric- choice is required (ctrl-c to quit)')
                 continue
             uin = int(uin)
             if uin not in idx:
-                print 'a choice between 1 and %s is required (ctrl-c to quit)' % len(choices)
+                print('a choice between 1 and %s is required (ctrl-c to quit)' % len(choices))
                 continue
 
             # all good,
             return idx[uin]
 
     except KeyboardInterrupt:
-        print
+        print()
         sys.exit(1)
 
 
-from contextlib import contextmanager
-import tempfile
-import shutil
```

Comment:
weird, good to move the imports at the top

Reply:
actually, you can thank a bumped version of pylint/autopep8 for that :)